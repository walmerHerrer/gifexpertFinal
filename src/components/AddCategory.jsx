import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    // setCategories( categories => [ inputValue, ...categories ]);
    setInputValue("");
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <div className="input-group">
        <input
          type="text"
          className="form-control"
          placeholder="Buscar gifs"
          value={inputValue}
          onChange={onInputChange}
          aria-describedby="button-addon2"
        />
        <button
          className="btn btn-success"
          type="button"
          id="button-addon2"
          onClick={onSubmit}
        >
          BUSCAR
        </button>
      </div>
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
