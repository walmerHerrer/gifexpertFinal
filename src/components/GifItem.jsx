import PropTypes from "prop-types";
import Col from "react-bootstrap/Col";

export const GifItem = ({ title, url, id }) => {
  return (
    <>
      <div className="col-3">
        <div className="card">
          <img src={url} alt={title} width={250} height={250} />
          <div className="card-content">
            <h2>{title}</h2>
            <a href="#">Leer m√°s</a>
          </div>
        </div>
      </div>
    </>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
