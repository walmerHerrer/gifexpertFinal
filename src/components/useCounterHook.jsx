import { useCounter } from "../hooks/useCounter";

export const CounterWithCustomHook = () => {
  const { counter, increment, decrement, reset } = useCounter(2);

  return (
    <>
      <h1>Counter with Hook: {counter}</h1>
      <hr />

      <button onClick={() => increment(3)} className="btn btn-primary">
        +3
      </button>
      <button onClick={reset} className="btn btn-primary">
        Reset
      </button>
      <button onClick={() => decrement(1)} className="btn btn-primary">
        -1
      </button>
    </>
  );
};
