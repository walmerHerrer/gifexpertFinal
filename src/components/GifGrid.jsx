import PropTypes from "prop-types";

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import Spinner from "react-bootstrap/Spinner";
import Row from "react-bootstrap/Row";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <div className="row">
        {isLoading && (
          <div className="d-flex justify-content-center">
            <div className="spinner-border" role="status">
              <span className="visually-hidden">Cargando...</span>
            </div>
          </div>
        )}
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
