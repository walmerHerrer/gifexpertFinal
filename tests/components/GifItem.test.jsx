import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GifItem />", () => {
  const title = "goku";
  const url = "https://one-punch.com/goku.jpg";
  //const url = "https://tenor.com/view/goku-jump-transformation-gif-22164699";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
    //expect(titleElement).toBeInTheDocument(); // Asegura que el elemento est√© en el DOM
  });
});
