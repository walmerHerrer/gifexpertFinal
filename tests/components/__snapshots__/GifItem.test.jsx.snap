// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="col-3"
  >
    <div
      class="card"
    >
      <img
        alt="goku"
        height="250"
        src="https://one-punch.com/goku.jpg"
        width="250"
      />
      <div
        class="card-content"
      >
        <h2>
          goku
        </h2>
        <a
          href="#"
        >
          Leer m√°s
        </a>
      </div>
    </div>
  </div>
</div>
`;
